src/MeshlessFV/MfvMuscl.cpp

```diff
@@ -103,23 +103,21 @@ void MfvMuscl<ndim, kernelclass,SlopeLimiter>::ComputeGodunovFlux
   FLOAT gradW[nvar][ndim];             // Gradient of primitive vector
   FLOAT dW[nvar];                      // Change in primitive quantities
   const FLOAT dt = timestep*(FLOAT) part.nstep;    // Timestep of given particle
-
-
-  FLOAT invh_i   = 1/part.h;
-  FLOAT volume_i = 1/part.ndens;
+  const FLOAT invh_i   = 1.0/part.h;
+  const FLOAT volume_i = 1.0/part.ndens;
 
   // Loop over all potential neighbours in the list
   //-----------------------------------------------------------------------------------------------
   for (jj=0; jj<Nneib; jj++) {
     j = neiblist[jj];
 
-    FLOAT invh_j   = 1/neibpart[j].h;
-    FLOAT volume_j = 1/neibpart[j].ndens;
+    const FLOAT invh_j   = (FLOAT) 1.0/neibpart[j].h;
```

Comment:
I hate this. It replaces a single character that is correct with unnecessary and ugly clutter. c++ will automatically cast the into to the smallest floating point type compatible with the others in the calculation.

That means if there are no doubles in the expression, the computation will be done in single precision, otherwise it will be done in double precision. The cast will be done at compile time.  See [here](http://stackoverflow.com/questions/5563000/implicit-type-conversion-rules-in-c-operators). I accept that if the coefficient was 0.5 then the cast is necessary to ensure the calculation is done in single precision.